### ISSUE TITLE
FITD Exceptions

### ISSUE THREAD

**Issue by ghost** [Sammi-Husky/Sm4sh-Tools#63]
When decompiling with FITD, I get OutOfRange for 4 fighters and KeyNotFound for D3

The output is the exact same for each fighter on each decompile

Bayonetta:
```
Unhandled Exception: System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeLoop(Int32 startIndex, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeCommand(Int32 index, UInt32 ident, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.Deserialize()
   at FITD.Program.write_movedef(ACMDScript game, ACMDScript effect, ACMDScript sound, ACMDScript expression, String animname, Boolean unlisted, String outdir)
   at FITD.Program.decompile_acmd(String mtable, String motionFolder, String output)
   at FITD.Program.Main(String[] args)
```

D3:
```
Unhandled Exception: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at SALT.Scripting.AnimCMD.ACMDCommand.get_ParamSpecifiers()
   at SALT.Scripting.AnimCMD.ACMDFile.ParseEventList(UInt32 cRC, Int32 offset)
   at SALT.Scripting.AnimCMD.ACMDFile..ctor(DataSource source)
   at FITD.Program.decompile_acmd(String mtable, String motionFolder, String output)
   at FITD.Program.Main(String[] args)
```

G&W
```
Unhandled Exception: System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeLoop(Int32 startIndex, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeCommand(Int32 index, UInt32 ident, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.Deserialize()
   at FITD.Program.write_movedef(ACMDScript game, ACMDScript effect, ACMDScript sound, ACMDScript expression, String animname, Boolean unlisted, String outdir)
   at FITD.Program.decompile_acmd(String mtable, String motionFolder, String output)
   at FITD.Program.Main(String[] args)
```

Palutena
```
Unhandled Exception: System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeLoop(Int32 startIndex, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeCommand(Int32 index, UInt32 ident, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.Deserialize()
   at FITD.Program.write_movedef(ACMDScript game, ACMDScript effect, ACMDScript sound, ACMDScript expression, String animname, Boolean unlisted, String outdir)
   at FITD.Program.decompile_acmd(String mtable, String motionFolder, String output)
   at FITD.Program.Main(String[] args)
```

Robin(reflet)
```
Unhandled Exception: System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeConditional(Int32 startIndex, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeCommand(Int32 index, UInt32 ident, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeConditional(Int32 startIndex, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeCommand(Int32 index, UInt32 ident, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeLoop(Int32 startIndex, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.DeserializeCommand(Int32 index, UInt32 ident, List`1& lines)
   at SALT.Scripting.AnimCMD.ACMDScript.Deserialize()
   at FITD.Program.write_movedef(ACMDScript game, ACMDScript effect, ACMDScript sound, ACMDScript expression, String animname, Boolean unlisted, String outdir)
   at FITD.Program.decompile_acmd(String mtable, String motionFolder, String output)
   at FITD.Program.Main(String[] args)
```

**Comment by Sammi-Husky** [comment]
Was this on the very latest commit build? Changes i made not too long ago fixed DDD decompiling for me and other testers so maybe double check with a redownload of the latest build?

**Comment by jugeeya** [comment]
These characters are actually fixed for me (I did have this exact problem on a much older version of FITX) but I still have this issue with the following characters: Bowser Jr., Robin, and Sonic. I am on the very latest commit (bda1f49). I have included the pictures to display the moves that are likely causing the problems. (the ones in parentheses are because I don't know if it successfully decompiled the move and breaks on the next one, or breaks on the one it says it's decompiling)

Bowser Jr.: SpecialSSpin.acm (or SpecialSSpin_C2.acm) 
![image](https://user-images.githubusercontent.com/15069495/28558756-238cb334-70c8-11e7-9061-a9723e661412.png)
Robin: SpecialLwStart.acm (or SpecialLwStart_C2.acm)
![image](https://user-images.githubusercontent.com/15069495/28558717-fb5719a4-70c7-11e7-8c3a-286fe3b3e874.png)
Sonic: 0x6AD985ED.acm (or 0x06679C9C.acm)
![image](https://user-images.githubusercontent.com/15069495/28558734-0ccde294-70c8-11e7-8c29-bce3b7cf4bcb.png)
